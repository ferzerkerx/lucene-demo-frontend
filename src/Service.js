import {checkStatus, parseJSON} from './Utils'

export const Service = (text) => {
    const encodedQuery = encodeURIComponent(text);
    return fetch(`http://localhost:8080/books?keyword=${encodedQuery}`, {credentials: 'include'})
        .then(checkStatus)
        .then(parseJSON)
        .catch((error)  => console.error('request failed', error));
};