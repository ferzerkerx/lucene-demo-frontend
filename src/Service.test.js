import {Service} from './Service'

beforeEach(function() {

  global.fetch = jest.fn().mockImplementation(() => {
    var p = new Promise((resolve, reject) => {
      resolve({
        ok: true,
        status: 200,
        Id: '123',
        json: function() {
          return {Id: '123'}
        }
      });
    });

    return p;
  });

});


it("searches the specified test", async function() {
  const response = await Service('bookName');
  expect(response).toMatchSnapshot();
});