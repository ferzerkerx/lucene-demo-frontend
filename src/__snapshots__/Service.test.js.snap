// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searches the specified test 1`] = `
Object {
  "Id": "123",
}
`;
